kasa/smart/smartdevice.py

```diff
@@ -104,14 +105,25 @@ async def _negotiate(self):
         We fetch the device info and the available components as early as possible.
         If the device reports supporting child devices, they are also initialized.
         """
-        initial_query = {"component_nego": None, "get_device_info": None}
+        initial_query = {
+            "component_nego": None,
+            "get_device_info": None,
+            "get_connect_cloud_state": None,
+        }
         resp = await self.protocol.query(initial_query)
 
         # Save the initial state to allow modules access the device info already
         # during the initialization, which is necessary as some information like the
         # supported color temperature range is contained within the response.
         self._last_update.update(resp)
         self._info = self._try_get_response(resp, "get_device_info")
+        # It's likely that all SMART devices respond to get_connect_cloud_state but
+        # just in case some don't pass a default {} here. The KP125M fixture does not
+        # have it because it was created prior to dump_devinfo having the response.
+        self._is_cloud_connected = bool(
+            (cloud_state := self._try_get_response(resp, "get_connect_cloud_state", {}))
+            and cloud_state["status"] == 0
+        )
```

Comment:
How about having this as a property that accesses the cloudmodule? Something like:
```
@property
def is_cloud_connected(self):
    if "CloudModule" not in self.modules: return False
    return self.modules["CloudModule"].is_connected()
```

```

Reply:
That would mean we need to ensure the `CloudModule` is loaded before the `Fimware` module which could be a bit flaky.

Reply:
As it is queried as part of the initial query, we could always initialize it and handle the error cases inside the module?

Reply:
Done